src/sentry/static/sentry/app/components/issueSyncListElement.jsx

```diff
@@ -3,60 +3,101 @@ import React from 'react';
 import styled from 'react-emotion';
 import InlineSvg from 'app/components/inlineSvg';
 import space from 'app/styles/space';
+import {capitalize} from 'lodash';
 
 class IssueSyncElement extends React.Component {
   static propTypes = {
     externalIssueLink: PropTypes.string,
     externalIssueId: PropTypes.string,
+    openModal: PropTypes.func,
     onClose: PropTypes.func,
-    integrationType: PropTypes.oneOf(['github', 'jira', 'vsts']),
-  };
-
-  static defaultProps = {
-    extenalIssueLink: '#',
+    integrationType: PropTypes.string,
   };
 
   isLinked() {
     return this.props.externalIssueLink && this.props.externalIssueId;
   }
 
+  handleDelete = evt => {
+    return this.props.onClose(this.props.externalIssueId);
+  };
+
+  getPrefix() {
+    switch (this.props.integrationType) {
+      case 'github':
+        return 'GH-';
+      case 'github_enterprise':
+        return 'GHE-';
+      default:
+        return this.getPrettyName() + '-';
+    }
+  }
+
   getIcon() {
     switch (this.props.integrationType) {
       case 'github':
         return <IntegrationIcon src="icon-github" />;
+      case 'github_enterprise':
+        return <IntegrationIcon src="icon-github" />;
       case 'jira':
         return <IntegrationIcon src="icon-jira" />;
       case 'vsts':
         return <IntegrationIcon src="icon-vsts" />;
       default:
-        return null;
+        return <IntegrationIcon src="icon-generic-box" />;
     }
   }
 
-  getText() {
-    return this.isLinked() ? (
-      <IntegrationLink href={this.props.externalIssueLink}>
-        {this.props.externalIssueId}
-      </IntegrationLink>
-    ) : (
-      <IntegrationLink href={'#'}>
-        Link <IntegrationName>{this.props.integrationType}</IntegrationName> Issue
+  getPrettyName() {
```

Comment:
Shouldn't this be handled on integration side?

Reply:
(btw I have little context on this so not saying you change it)

Reply:
maybe it should be? But I don't think we do it anywhere atm. I think this might be a good thing to follow up on

Reply:
Agreed

Reply:
Yeah this should definitely be handled on the integration side. We want to avoid having frontend code that's specific to one integration since it makes it hard to add new integrations, especially since we don't have any hooks for adding frontend stuff to integrations atm